Feature: Internationalization App

peterampazzo opened the pull request:
This PR includes the integration of `i18next` a complete solution to localize applications.
The supported languages are English (as default) and Italian. 

It's possibile to detect user language, all the text will be displayed in the same language if it's supported. The fallback language remains English.

This features was mentioned on issue #19.

andtos90 commented on src/i18n.js:
```diff
@@ -0,0 +1,21 @@
+import { initReactI18next } from "react-i18next";
+
+i18n
+  .use(Backend)
```
It is probably easier and has more sense to load translations directly in from json files than doing a separate http request. Keep in mind we have a really small set of translations

andtos90 commented on src/Root.js:
```diff
@@ -11,7 +11,9 @@ function Root() {
       <CssBaseline />
       <StateProvider>
-        <App />
+        <Suspense>
```
Do we need this without i18n Provider?

peterampazzo commented:
Thank you for your advice. To sum up: 
* I moved `/public/locales` into `/src/locales`
* I removed `i18next-http-backend` so now you directly include the two json - and also remove `Suspense`
* I added a button which allows users (after joined an event) to change the language - it'll appear a tooltip with the original name of the other available language

**ENGLISH**
<img width="1440" alt="Screenshot 2020-04-15 at 22 19 57" src="https://user-images.githubusercontent.com/4621567/79384885-f8c15400-7f67-11ea-951e-4246f91b4caf.png">

**ITALIAN**
<img width="1440" alt="Screenshot 2020-04-15 at 22 19 44" src="https://user-images.githubusercontent.com/4621567/79384886-f9f28100-7f67-11ea-93b8-e551daa773ca.png">
